ess/src/API/EMBC.ESS/Managers/Submissions/SubmissionsManager.cs

```diff
@@ -310,15 +310,22 @@ public async Task<VerifySecurityPhraseResponse> Handle(VerifySecurityPhraseQuery
 
         public async Task<string> Handle(SaveEvacuationFileNoteCommand cmd)
         {
+            if (string.IsNullOrEmpty(cmd.Note.Id) && string.IsNullOrEmpty(cmd.FileId)) throw new ArgumentNullException("NoteId and FileId cannot both be empty/null.");
```

Comment:
it is better to split the checks to different exceptions (better clarity)

Reply:
The idea is that if note id is null then we are creating a new note and the file id is needed to link the note.
If note id is populated then the file id isn't needed.

So the only real bad condition is if they are both empty.

Reply:
wouldn't the file ID be mandatory and the not id will determine if this is update or insert? then you can always enforce the right file is linked to the note, and won't be changed in an update